server/service/http/message/query/query_structure.rs

```diff
@@ -278,6 +265,9 @@ fn query_structure_edge(
         | Constraint::Value(_)
         | Constraint::Is(_)
         | Constraint::Iid(_) => {}
+        Constraint::IndexedRelation(indexed) => {
+            unreachable!("This is unreachable since we extract query-structure before we apply transformations");
+        }
```

Comment:
I think we should keep the handling as it was. It was already implemented, and I can totally see us pushing the shortcut edges to initial translation at some point.